lib/Doctrine/Common/Collections/Expr/ClosureExpressionVisitor.php

```diff
@@ -41,6 +41,10 @@ public static function getObjectFieldValue($object, string $field)
             return $object[$field];
         }
 
+        if (method_exists($object, $field)) {
```

Comment:
`method_exists` also returns `true` if the method is not public.
https://3v4l.org/LAgYk

Reply:
Exactly the same as it works without my changes